sdcm/cluster_aws.py

```diff
@@ -340,7 +340,8 @@ def update_bootstrap(ec2_user_data, enable_auto_bootstrap):
         return ec2_user_data
 
     def _create_or_find_instances(self, count, ec2_user_data, dc_idx, az_idx=0):
-        if self.nodes:
+        nodes = [node for node in self.nodes if node.dc_idx == dc_idx]
```

Comment:
This is exactly solution that I suggested to do. @dkropachev suggested more advance solution that can cover more cases

Reply:
Since we assume the issue is only with multi-dc cases, I thought we have time to try a better soultion @dkropachev suggest.

but we were missing the root casue that made it start happening, and that was @soyacz AZ related changes.

Reply:
now it's become more urgent, since it's effect all cases, not just multi-region ones, and we need quick fix (less complicated, at least for now)

Reply:
> Since we assume the issue is only with multi-dc cases, I thought we have time to try a better soultion @dkropachev suggest.
> 
> but we were missing the root casue that made it start happening, and that was @soyacz AZ related changes.

After code analysis I think this was happening all the time for multi dc, AZ support only made it worse.

Reply:
Why `az_idx` is not part of the condition?

Reply:
@soyacz ^ ?

Reply:
for now when having multi-az we don't follow this logic as provision step is disabled. When adapting provision step to support multi-az then it must be implemented. I already work on it. (currently we don't support filtering by az when getting ec2 instances)

Reply:
@soyacz thanks for the explanation, I think, it would be really useful to have it as a comment in the code.